### DISCUSSION TITLE
Ad hoc query filters currently only have a string field, and no metadata. Add metadata

### DISCUSSION THREAD

**Discussion by srclosson** [grafana/grafana#38257] [Dashboards, Panels & Visualizations]
**What would you like to be added**:

If I have an adhoc filter, currently it is only a string representation and there is no metadata associated.

For example: In the plugin development and ui libraries we have a "SelectableValue" which has a label, and then a value where we can associate metadata. This is ideal.

I would like to see the adhoc filters use something similar.

**Why is this needed**:
An issue I run into is where a customer api always associates data in key/value pairs. For example:

A hostname is under the key 'Host', and the value 'HOST'. When querying the api, I must use 'HOST'. 

When I get an adhoc filter where
Host = SomeHostName

I have to do the folowing:
resolved `host` to `HOST`
resolve `SomeHostName` to `SOME-VALUE-FOR-HOST`

then query `HOST,SOME-VALUE-FOR-HOST`.

This however could have been associated with the key in the adhoc fiter. Ie:
```
 {
    label: "Host",
    value: "HOST",
 }
```

This would save a ton of unnecessary asynchronous calls and improve plugin performance.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
